### DISCUSSION TITLE
Are there any "patterns" for things like autocomplete, type-ahead and similars?

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#391] [Q&A]
Since Unpoly doesn't have the same strategy htmx does (partials), these type of behaviours are more complex to deal with. Do you guys have any suggestions, patterns or JS libs to share?

**Comment by adam12** [comment]
Did you have an example of an autocomplete/type-ahead that you're looking to implement?

**Reply by alexandremjacques** [reply]
Nothing really specific. I'm just trying to figure out some patterns for Unpoly.

E.g. right now I have a use case for searcing for customers by name and pick one from the matched list. I guess I could use a simple list-under-the-input and click on the desired match.

I guess a simple pattern on how to deal with these kind of things would be nice to have (since the're pretty much the same - retrieve a separate list of things - not inside the whole page response - and present to the user).
